### PR TITLE
Add doc and test for the fact that a quantifier's bindings can be empty

### PR DESCRIPTION
**Pull request by wilcoxjay**
This PR adds a bit of documentation and a test to call out the fact that a `Term::Quantifier`'s list of bindings can be empty.

Not saying I think it *should* work this way, but this is how the code works. The parser supports it, and at least some paths in the code (eg [`semantics::eval_assign`](https://github.com/vmware-research/temporal-verifier/blob/36e811450c1ecc3841ee2cc56f016d16d66cad9b/fly/src/semantics.rs#L271-L275)) handle this case.

### CODE REVIEW

File: fly/src/syntax.rs [hunk 1]
```diff
@@ -97,6 +97,7 @@ pub enum Term {
     #[allow(missing_docs)]
     Quantified {
         quantifier: Quantifier,
+        /// The sequence of bindings bound by this quantifier. Might be empty!
```
Comment (tchajed) [inline fly/src/syntax.rs:100]:
```suggestion
        /// The sequence of bindings bound by this quantifier. Might be empty.
```

**Review by tchajed — COMMENTED** [review]
I'm also not sure if this is what we want, but it seems fine. Some simplification pass would ideally get rid of these before it reaches most of the code.

File: fly/src/syntax.rs (hunk 1 above)
Comment (tchajed) [reply fly/src/syntax.rs:100]: I don't think empty quantifiers deserve _two_ exclamation marks 😄
